File: semestr-8/Jezyk C#/alexander/Csharp/Archiwum/2021s/Id06IO2/Strategy/Strategy/Sportsman.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Strategy
{
    public class Sportsman
    {
        private string name;
        private StrategyRun strun;
        private StrategyJump stjump;
        private StrategySwim stswim;
        public Sportsman(string name, StrategyRun strun, StrategyJump stjump, StrategySwim stswim) 
        {
            this.name=name;
            this.strun = strun;
            this.stjump = stjump;
            this.stswim = stswim;
        }
        public string Run() { return strun.execute(); }
        public string Jump() { return stjump.execute(); }
        public string Swim() { return stswim.execute(); }
        public string execute(SportsmenCommand cmd) { return name+": "+cmd.execute(this); }
    }
}

File: semestr-8/Programowanie obiektowe w Java/alexander/POJ/Archiwum/2018s/Command/src/Sportsman.java
public class Sportsman 
{
   private String name;
   private final RunStrategy strun;
   private final JumpStrategy stjump;
   private final SwimStrategy stswim;
   public Sportsman(String name,RunStrategy strun,JumpStrategy stjump,SwimStrategy stswim)
   {
      this.name=name;
      this.strun=strun;
      this.stjump=stjump;
      this.stswim=stswim;
   }
   public String getName() { return name; }
   public String run() { return strun.execute(); }
   public String jump() { return stjump.execute(); }
   public String swim() { return stswim.execute(); }
}